src/template.rs

```diff
@@ -0,0 +1,67 @@
+use handlebars::{handlebars_helper, Handlebars};
+use rand::seq::SliceRandom;
+
+pub fn addhelpers(x: &mut Handlebars) {
+    handlebars_helper!(wrap: |index: usize, s: String| {
+        wrap_helper(index, "/n".to_string(), s)
+    });
+    handlebars_helper!(wrap_with: |index: usize, ending: String, s: String| {
+        wrap_helper(index, ending, s)
+    });
+    handlebars_helper!(contains: |sub_str: String, content: String | content.contains(&sub_str));
+    handlebars_helper!(has_prefix: |prefix: String, content: String | content.starts_with(&prefix));
+    handlebars_helper!(has_suffix: |suffix: String, content: String | content.ends_with(&suffix));
+    handlebars_helper!(cat: |s: Vec<String>| s.join(" "));
+    handlebars_helper!(replace: |target: String, replacememt: String, s: String| s.replace(&target, &replacememt));
+    handlebars_helper!(shuffle: |s: String| {
+        let mut bytes = s.into_bytes();
+        let mut rng = rand::thread_rng();
+        bytes.shuffle(&mut rng);
+        let str = String::from_utf8(bytes).unwrap();
+        str
+    });
+    handlebars_helper!(indent: |count: usize, s: String| {
+        let mut data = s.clone();
+        for _ in 0..count {
+            data.insert(0, ' ');
+        }
+        data
+    });
+
+    x.register_helper("wrap", Box::new(wrap));
+    x.register_helper("wrap_with", Box::new(wrap_with));
+    x.register_helper("contains", Box::new(contains));
+    x.register_helper("has_prefix", Box::new(has_prefix));
+    x.register_helper("has_suffix", Box::new(has_suffix));
+    x.register_helper("cat", Box::new(cat));
+    x.register_helper("indent", Box::new(indent));
+    x.register_helper("replace", Box::new(replace));
+    x.register_helper("shuffle", Box::new(shuffle));
+}
+
+fn wrap_helper(index: usize, ending: String, s: String) -> String {
+    let mut data = s.clone();
+    let mut wraploc: usize = 0;
+    let mut lastwrap: usize = 0;
+    let mut i = 0;
+
+    while i < s.len() - 1 {
+        wraploc += 1;
+        if wraploc >= index {
```

Comment:
i find the code more readable when there are not too many nested if conditions. you can probably write this as:

```
if wraploc > index {
    i += 1
    continue
}

for k in (0..i).rev() 
.
.
.